src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -838,7 +838,7 @@ internal virtual void ValidateParameters()
         internal virtual void PrepareSession()
         {
             // Make sure we have a valid WebRequestSession object to work with
-            WebSession ??= new WebRequestSession();
+            // WebSession ??= new WebRequestSession();
```

Comment:
I think it is not good to allocate if we have no need to object.
Here we could think about MemberNotNullAttribute.

Reply:
I think [System.Diagnostics.CodeAnalysis.AllowNullAttribute] could be the right choice; WebSession is never null

Reply:
If it is always not null then it really can be initialized at the definition point

Reply:
Actually it seems we can't inizialize WebSession at the definition point without changing line 697, and I'd rather not do it
```cs
            if (WebSession is not null && SessionVariable is not null)
            {
                ErrorRecord error = GetValidationError(WebCmdletStrings.SessionConflict, "WebCmdletSessionConflictException");
                ThrowTerminatingError(error);
            }
```

Reply:
Intention of the check is that these parameters are mutually exclusive. So we could check `presence` of these parameters.